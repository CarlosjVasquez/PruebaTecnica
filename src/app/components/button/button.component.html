<button
  [type]="type"
  class="btn btn-{{ color }} btn-{{ size }}"
  [class.is-btn-disabled]="disabled"
  (click)="onHandleClick()"
  [disabled]="disabled"
>
  {{ label }}
</button>
