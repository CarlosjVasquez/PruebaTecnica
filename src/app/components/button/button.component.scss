@import "../../../styles/variables";

.btn {
    display: inline-block;
    font-weight: 600;
    text-align: center;
    white-space: nowrap;
    vertical-align: middle;
    user-select: none;
    border: 1px solid transparent;
    padding: 0.7rem 1rem;
    font-size: 0.8rem;
    line-height: 1.5;
    border-radius: 0.25rem;
    transition: color 0.1s ease-in-out, background-color 0.1s ease-in-out,
    border-color 0.1s ease-in-out, box-shadow 0.1s ease-in-out;
  
    &-sm {
      padding: map-get($btn-sizes, sm);
      font-size: 0.875rem;
    }
  
    &-md {
      padding: map-get($btn-sizes, md);
      font-size: 1rem;
    }
  
    &-lg {
      padding: map-get($btn-sizes, lg);
      font-size: 1.25rem;
    }
  }
  
  .is-btn-disabled {
    pointer-events: none;
    opacity: 0.5;
    cursor: not-allowed;
  }