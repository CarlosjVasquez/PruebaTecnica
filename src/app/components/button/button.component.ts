import { Component, EventEmitter, Input, Output } from '@angular/core';

@Component({
  selector: 'custom-button',
  standalone: true,
  imports: [],
  templateUrl: './button.component.html',
  styleUrl: './button.component.scss',
})
export class ButtonComponent {
  @Input() color: string = 'primary';
  @Input() size: 'sm' | 'md' | 'lg' = 'md';
  @Input() label: string = 'Button';
  @Input() disabled: boolean | null = false;
  @Input() type: 'button' | 'menu' | 'reset' | 'submit' = 'button';
  @Output() onEventClick: EventEmitter<void> = new EventEmitter();

  onHandleClick() {
    this.onEventClick.emit();
  }
}
