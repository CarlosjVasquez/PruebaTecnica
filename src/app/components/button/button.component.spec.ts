import { ComponentFixture, TestBed } from '@angular/core/testing';

import { ButtonComponent } from './button.component';
import { By } from '@angular/platform-browser';

describe('ButtonComponent', () => {
  let component: ButtonComponent;
  let fixture: ComponentFixture<ButtonComponent>;
  let compiled: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      providers: [ButtonComponent],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(ButtonComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  test('should create', () => {
    expect(component).toBeTruthy();
  });

  test('match to snapshot', () => {
    expect(compiled).toMatchSnapshot();
  });

  test('should emit event click', () => {
    jest.spyOn(component.onEventClick, 'emit');

    const buttonElement = fixture.debugElement.query(
      By.css('button')
    ).nativeElement;
    buttonElement.click();

    expect(component.onEventClick.emit).toHaveBeenCalled();
  });
});
