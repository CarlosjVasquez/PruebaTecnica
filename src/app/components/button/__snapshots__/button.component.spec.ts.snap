// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonComponent match to snapshot 1`] = `
<div
  id="root1"
>
  <button
    class="btn btn-md btn-primary"
    type="button"
  >
     Button

  </button>
</div>
`;
